### PR TITLE
Added systemtap testsuite_v1.

### PR DESCRIPTION
**Pull request by poojabs7**
This test runs the systemtap testsuite.

Signed-off-by: pooja <pooja@linux.vnet.ibm.com>

### CODE REVIEW

File: toolchain/systemtap.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Systemtap(Test):
+
+    def setUp(self):
+        smm = SoftwareManager()
+        packages = ['make', 'gcc', 'systemtap', 'systemtap-runtime',
+                    'elfutils', 'kernel-devel', 'kernel-debug',
+                    'kernel-debuginfo', 'kernel-debug-debuginfo', 'dejagnu',
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: i think for ubuntu kernel-debuginfo package name  will be different, can you please check

File: toolchain/systemtap.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                                    shell=True)
+        self.log.info(script_result)
+        if script_result.exit_status != 0:
+            self.fail("simple systemtap test failed,"
+                      "kernel debuginfo package may be missing")
+
+    def test(self):
+        build.make(self.srcdir, extra_args='installcheck')
+        failed_tests = process.system_output("grep -w FAIL systemtap.sum",
+                                             shell=True, ignore_status=True)
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: ok

File: toolchain/systemtap.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                                    " }'", ignore_status=True, sudo=True,
+                                    shell=True)
+        self.log.info(script_result)
+        if script_result.exit_status != 0:
+            self.fail("simple systemtap test failed,"
+                      "kernel debuginfo package may be missing")
+
+    def test(self):
+        build.make(self.srcdir, extra_args='installcheck')
+        failed_tests = process.system_output("grep -w FAIL systemtap.sum",
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: can we also extend  this test to check user input probe file ? just a suggestion

File: toolchain/systemtap.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                                             shell=True, ignore_status=True)
+        if failed_tests:
+            self.log.info(failed_tests)
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: i think` failed_tests `is indicated number right ? where i find log is it part of avocado log dir ? i ask this if you say check the log i think systemtap.sum is log file can you please mention as place a comment in test file about detailed path

File: toolchain/systemtap.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        build.make(self.srcdir, extra_args='install')
+        self.test_dir = os.path.join(self.srcdir, "testsuite")
+        os.chdir(self.test_dir)
+        # Run a simple systemtap script to make sure systemtap and the
+        # kernel debuginfo packages are correctly installed
+        script_result = process.run("PATH=%s/bin:$PATH stap -v /bin/true "
+                                    "-e 'probe vfs.read { exit()"
+                                    " }'", ignore_status=True, sudo=True,
+                                    shell=True)
+        self.log.info(script_result)
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: i think this is kind of sanity check for test so i think we can remove this info right ?

File: toolchain/systemtap.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        if script_result.exit_status != 0:
+            self.fail("simple systemtap test failed,"
```
Comment (PraveenPenguin) [inline toolchain/systemtap.py]: please skip the test as test is not able to run

**Comment by poojabs7** [comment]
Raised New PR.

https://github.com/avocado-framework-tests/avocado-misc-tests/pull/412

**Pull request closed without merging**